src/utils/hooks/transaction/utils/fee.ts

```diff
@@ -119,17 +119,44 @@ const wrapWithTxFeeSwap = (
   return { extrinsic: wrappedCall };
 };
 
-const getActionAmount = (params: TransactionActions): MonetaryAmount<CurrencyExt> | undefined => {
+const getActionAmount = (
+  params: TransactionActions,
+  feeCurrency: CurrencyExt
+): MonetaryAmount<CurrencyExt> | undefined => {
+  let amounts: MonetaryAmount<CurrencyExt> | MonetaryAmount<CurrencyExt>[] | undefined;
+
   switch (params.type) {
     case Transaction.TOKENS_TRANSFER: {
       const [, amount] = params.args;
-      return amount;
+      amounts = amount;
+      break;
     }
+    /* START - AMM */
     case Transaction.AMM_SWAP: {
       const [trade] = params.args;
-      return trade.inputAmount;
+      amounts = trade.inputAmount;
+      break;
+    }
+    case Transaction.AMM_ADD_LIQUIDITY: {
+      const [pooledAmounts] = params.args;
+      amounts = pooledAmounts;
+      break;
     }
+    case Transaction.AMM_REMOVE_LIQUIDITY: {
+      const [amount] = params.args;
+      amounts = amount;
+      break;
+    }
+    /* END - AMM */
   }
+
+  if (!amounts) return;
+
+  if (Array.isArray(amounts)) {
+    return amounts.find((amount) => isCurrencyEqual(amount.currency, feeCurrency));
+  }
+
+  return isCurrencyEqual(amounts.currency, feeCurrency) ? amounts : undefined;
```

Comment:
I would suggest to always define amounts as array to be have more clarity about the value and simpler return statement without conditions.